lib/kaufmann_ex/test_support/mock_schema_registry.ex

```diff
@@ -29,6 +29,18 @@ defmodule KaufmannEx.TestSupport.MockSchemaRegistry do
   end
 
   @spec encode_decode(Request.t()) :: Event.t()
+  def encode_decode(%Request{format: format} = request) when format == :json or format == :avro do
```

Comment:
`format` should always default to `:default`, so we could eliminate the brute force encode/decode completely as well as the guards for `:json` and `:avro` here.